File: samples/client/wordnik-api/python/tests/WordListApiTest.py
#!/usr/bin/env python

import sys
import unittest
import urllib2
import json
from pprint import pprint
from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from wordnik import *


class WordListApiTest(BaseApiTest):

    def setUp(self):
        super(WordListApiTest, self).setUp()
        self.authToken = self.accountApi.authenticate(self.username,
                                                      self.password).token
        self.existingList = self.accountApi.getWordListsForLoggedInUser(self.authToken,
                                                                        limit=1)[0]

        from wordnik.models import WordList
        wordList = WordList.WordList()
        wordList.name = "my test list"
        wordList.type = "PUBLIC"
        wordList.description = "some words I want to play with"

        # sampleList = self.wordListApi.createWordList(wordList,
        #                                               self.authToken)

        # if not sampleList:
        #     raise Exception("can't create test list to run tests with")

    def testGetWordListByPermalink(self):
        res = self.wordListApi.getWordListByPermalink(self.existingList.permalink,
                                                      self.authToken)
        assert res, 'null getWordListByPermalink result'

    def testGetWordListByPermalink(self):
        res = self.wordListApi.getWordListByPermalink(self.existingList.permalink,
                                                      self.authToken)
        assert res, 'null getWordListByPermalink result'

    def testUpdateWordList(self):
        import time
        description = 'list updated at ' + str(time.time())
        self.existingList.description = description
        self.wordListApi.updateWordList(self.existingList.permalink,
                                        self.authToken, body=self.existingList)

        res = self.wordListApi.getWordListByPermalink(self.existingList.permalink,
                                                      self.authToken)

        assert res.description == description, 'did not update wordlist'

    def testAddWordsToWordList(self):

        from wordnik.models import StringValue
        wordsToAdd = []
        word1 = StringValue.StringValue()
        word1.word = "delicious"
        wordsToAdd.append(word1)
        word2 = StringValue.StringValue()
        word2.word = "tasty"
        wordsToAdd.append(word2)
        word3 = StringValue.StringValue()
        word3.word = "scrumptious"
        wordsToAdd.append(word3)
        self.wordListApi.addWordsToWordList(self.existingList.permalink,
                                        self.authToken, body=wordsToAdd)

        res = self.wordListApi.getWordListWords(self.existingList.permalink,
                                                self.authToken)
        listSet = set([word.word for word in res])
        addedSet = set(["delicious", "tasty", "scrumptious"])
        assert len(listSet.intersection(addedSet)) == 3, 'did not get added words'

    def testDeleteWordsFromList(self):
        from wordnik.models import StringValue
        wordsToRemove = []
        word1 = StringValue.StringValue()
        word1.word = "delicious"
        wordsToRemove.append(word1)
        word2 = StringValue.StringValue()
        word2.word = "tasty"
        wordsToRemove.append(word2)
        word3 = StringValue.StringValue()
        word3.word = "scrumptious"
        wordsToRemove.append(word3)
        self.wordListApi.deleteWordsFromWordList(self.existingList.permalink,
                                                 self.authToken,
                                                 body=wordsToRemove)

        res = self.wordListApi.getWordListWords(self.existingList.permalink,
                                                self.authToken)
        listSet = set([word.word for word in res])
        addedSet = set(["delicious", "tasty", "scrumptious"])
        assert len(listSet.intersection(addedSet)) == 0, 'did not get removed words'


if __name__ == "__main__":
    unittest.main()

File: samples/client/wordnik-api/php/tests/WordListApiTest.php
<?php

require_once 'BaseApiTest.php';

date_default_timezone_set('America/Los_Angeles');

class WordListApiTest extends BaseApiTest {

  public function setUp() {
    parent::setUp();
    $this->authToken = $this->accountApi->authenticate($this->username,
                                                      $this->password)->token;
    $lists = $this->accountApi->getWordListsForLoggedInUser($this->authToken, $skip=null, $limit=1);
    $this->existingList = $lists[0];

    $this->wordList = new WordList();
    $this->wordList->name = "my test list";
    $this->wordList->type = "PUBLIC";
    $this->wordList->description = "some words I want to play with";
  }


  public function testGetWordListByPermalink() {
    $res = $this->wordListApi->getWordListByPermalink($this->existingList->permalink,
                                           $this->authToken);
    $this->assertNotEquals(null, $res);
  }


  public function testUpdateWordList() {
    $description = 'list updated at ' . time();
    $this->existingList->description = $description;
    $this->wordListApi->updateWordList($this->existingList->permalink,
                                       $body=$this->existingList,
                                       $this->authToken);
    $res = $this->wordListApi->getWordListByPermalink($this->existingList->permalink, $this->authToken);
    $this->assertEquals($description, $res->description);
  }

  public function testAddWordsToWordList() {
    $wordsToAdd = array();
    $word1 = new StringValue();
    $word1->word = "delicious";
    $wordsToAdd[] = $word1;
    $word2 = new StringValue();
    $word2->word = "tasty";
    $wordsToAdd[] = $word2;
    $word3 = new StringValue();
    $word3->word = "scrumptious";
    $wordsToAdd[] = $word3;
    $word4 = new StringValue();
    $word4->word = "not to be deleted";
    $wordsToAdd[] = $word4;

    $this->wordListApi->addWordsToWordList($this->existingList->permalink,
                                           $body=$wordsToAdd,
                                           $this->authToken);

    $res = $this->wordListApi->getWordListWords($this->existingList->permalink, $this->authToken, $sortBy=null, $sortOrder=null, $skip=null, $limit=null);

    $returnedWords = array();

    foreach ($res as $wordListWord) {
      $returnedWords[] = $wordListWord->word;
    }

    $intersection = array();
    foreach ($wordsToAdd as $addedWord) {
      if (in_array($addedWord->word, $returnedWords)) {
        $intersection[] = $addedWord->word;
      }
    }

    $this->assertEquals(4, count($intersection));
  }


  public function testDeleteWordsFromList() {
    $wordsToRemove = array();
    $word1 = new StringValue();
    $word1->word = "delicious";
    $wordsToRemove[] = $word1;
    $word2 = new StringValue();
    $word2->word = "tasty";
    $wordsToRemove[] = $word2;
    $word3 = new StringValue();
    $word3->word = "scrumptious";
    $wordsToRemove[] = $word3;

    $res = $this->wordListApi->getWordListWords($this->existingList->permalink, $this->authToken, $sortBy=null, $sortOrder=null, $skip=null, $limit=null);

    $this->wordListApi->deleteWordsFromWordList($this->existingList->permalink,
                                           $body=$wordsToRemove,
                                           $this->authToken);

    $res = $this->wordListApi->getWordListWords($this->existingList->permalink, $this->authToken, $sortBy=null, $sortOrder=null, $skip=null, $limit=null);

    $returnedWords = array();
    foreach ($res as $wordListWord) {
      $returnedWords[] = $wordListWord->word;
    }

    $intersection = array();
    foreach ($wordsToRemove as $removedWord) {
      if (in_array($removedWord->word, $returnedWords)) {
        $intersection[] = $removedWord->word;
      }
    }

    $this->assertEquals(0, count($intersection));
  }


}
?>
